bot/cogs/moderation.py

```diff
@@ -1207,6 +1241,30 @@ def _infraction_to_string(self, infraction_object):
             if User in error.converters:
                 await ctx.send(str(error.errors[0]))
 
+    async def respect_role_hierarchy(self, ctx: Context, target: Member, infraction_type: str) -> bool:
+        """
+        Check if the highest role of the invoking member is less than or equal to the target member
+
+        If this check fails, a warning is sent to the invoking ctx
+
+        Implement as a method rather than a check in order to avoid having to reimplement parameter
+        checks & conversions in a dedicated check decorater
+        """
+
+        # Build role hierarchy
+        actor = ctx.author
+        role_hierarchy = {role: rank for rank, role in enumerate(reversed(ctx.guild.roles))}
+        hierarchy_check = role_hierarchy[actor.top_role] <= role_hierarchy[target.top_role]
```

Comment:
can't we just compare the top roles directly? see https://discordpy.readthedocs.io/en/rewrite/api.html#role

Reply:
Sorry, forgot about this.